main.cpp

```diff
@@ -65,7 +68,9 @@ tuple<void*, size_t> allocate_pic(const string& filename) {
   DWORD old_protection;
   auto prot_result = VirtualProtectEx(GetCurrentProcess(), pic, pic_size, PAGE_EXECUTE_READ, &old_protection);
   if (!prot_result) throw runtime_error("[-] Couldn't VirtualProtectEx: " + GetLastError());
-  return { pic, pic_size };
+  MyTuple t1((void*) pic, (size_t) pic_size );
```

Comment:
Please use static_cast<void*> / static_cast<size_t> and just return as an rvalue:

```cpp
return MyTuple(static_cast<void*>(pic), static_cast<size_t>(pic_size));
```